pkg/apis/core/v1/helper/helpers.go

```diff
@@ -413,6 +413,32 @@ func TolerationsTolerateTaintsWithFilter(tolerations []v1.Toleration, taints []v
 	return true
 }
 
+// FindMatchingUntoleratedTaint checks if the given tolerations tolerates
+// all the filtered taints, and returns the first taint without a toleration
+func FindMatchingUntoleratedTaint(taints []v1.Taint, tolerations []v1.Toleration, applyFilter taintsFilterFunc) *v1.Taint {
```

Comment:
nit: just return an object instead of a pointer.

Reply:
need to return nil, if no taint is found

Reply:
true!